### PR TITLE
Add reactA11yEventHasRoleRule and test for the rule.

### PR DESCRIPTION
**Pull request by t-ligu**

### CODE REVIEW

File: tests/ReactA11yEventHasRoleRuleTest.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    it('should pass when element is not a dom element', () => {
+        const script: string = `
+            import React = require('react');
+
+            const a = <Div onclick />;
+            const b = <DIV></DIV>;
+        `;
+        TestHelper.assertNoViolation(ruleName, script);
+    });
+    it('should pass when dom element with event handel has implicit role', () => {
```
Comment (ipip2005) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: typo `handel` -> `handler`

File: tests/ReactA11yEventHasRoleRuleTest.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        const script: string = `
+            import React = require('react');
+
+            const a = <a href='hrefValue' onclick />;
+        `;
+        TestHelper.assertNoViolation(ruleName, script);
+    });
+
+    it('should pass when dom element with event handel has role attribute', () => {
```
Comment (ipip2005) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: typo handel -> handler

File: tests/ReactA11yEventHasRoleRuleTest.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        const script: string = `
+            import React = require('react');
+
+            const a = <div onclick role />;
+            const b = <div click role></div>;
+        `;
+        TestHelper.assertNoViolation(ruleName, script);
+    });
+
+    it('should fail when dom element with event handel has no role attribute and implicit role', () => {
```
Comment (ipip2005) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: typo handel -> handler

File: src/reactA11yEventHasRoleRule.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    private checkJsxOpeningElement(node: ts.JsxOpeningElement): void {
+        const tagName: string = node.tagName.getText();
+
+        if (!DOM_SCHEMA[tagName]) {
+            return;
+        }
+
+        const attributes: { [propName: string]: ts.JsxAttribute } = getJsxAttributesFromJsxElement(node);
+        const events: string[] = TARGET_EVENTS.filter((eventName: string): boolean => !!attributes[eventName]);
+        const hasRoleAttribute: boolean = !!attributes[ROLE_STRING] || !!getImplicitRole(node);
```
Comment (t-ligu) [inline src/reactA11yEventHasRoleRule.ts]: Fixed, thanks.

File: tests/ReactA11yEventHasRoleRuleTest.ts (hunk 1 above)
Comment (t-ligu) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: Fixed, thanks.

File: tests/ReactA11yEventHasRoleRuleTest.ts (hunk 2 above)
Comment (t-ligu) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: Fixed, thanks.

File: tests/ReactA11yEventHasRoleRuleTest.ts (hunk 3 above)
Comment (t-ligu) [inline tests/ReactA11yEventHasRoleRuleTest.ts]: Fixed, thanks.

**Pull request closed without merging**